httpserver/onion_processing.cpp

```diff
@@ -317,8 +246,8 @@ void RequestHandler::process_onion_req(const std::string& ciphertext,
         // Forward the request to url but only if it ends in `/lsrpc`
         if ((target.rfind("/lsrpc") == target.size() - 6) &&
```

Comment:
This code has a subtle bug in that it admits any 5-character path.  Perhaps steal https://github.com/oxen-io/oxen-core/blob/3ae23986009c9f8652add870938b4d53110150ae/src/common/string_util.h#L33-L36 to use here?

Reply:
Updated